vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs

```diff
@@ -336,7 +336,10 @@ type internal FSharpGoToDefinitionService [<ImportingConstructor>]
                         return results.AsEnumerable()
                     else // we need to get an FSharpSymbol from the targetRange found in the signature
                          // that symbol will be used to find the destination in the corresponding implementation file
-                        let implFilePath = Path.ChangeExtension (sigDocument.FilePath,"fs")
+                        let implFilePath =
+                            // Bugfix: apparently sigDocument not always is a signature file
+                            if isSignatureFile sigDocument.FilePath then Path.ChangeExtension (sigDocument.FilePath, "fs") 
+                            else sigDocument.FilePath
```

Comment:
I wonder if it wouldn't be a better solution to just setup script projects and compile script files just like the rest.
Now they are treated in a special and ephemeral way and that causes problems.